### ISSUE TITLE
cmake, building casarest using hdf5 support, fails

### ISSUE THREAD

**Issue by ygrange** [casacore/casarest#17]
I try to build casarest with HDF5 support (I built the latest version of hdf5). To this end, I built casacore with the DHDF5_ROOT_DIR and DUSE_HDF5 flags. 

For the cmake of casarest, I use the DHDF5_ROOT_DIR  flag of cmake. The make crashes at almost 100% with 
```
CMakeFiles/lwimager.dir/apps/lwimager.cc.o: In function 
`casa::HDF5Group::HDF5Group(casa::HDF5Object const&, casa::String const&, bool, bool)':
lwimager.cc:
(.text._ZN4casa9HDF5GroupC2ERKNS_10HDF5ObjectERKNS_6StringEbb[_ZN4casa9HDF5GroupC5ERKNS_10HDF5ObjectERKNS_6StringEbb]+0x7a):
 undefined reference to `casa::HDF5Group::init(int, casa::String const&, casa::String const&, bool, bool)'
```

I don't really know how to fix this. Any ideas?

**Comment by oonkjbr** [comment]
Same here. Upon compiling (for centos7) casarest (v1.3.1) directly after casacore (v2.3.0) in a standard LOFAR compilation I get an error regarding lwimager and hdf5 in the make. Is there a solution? I have now also tried casarest v1.4.2 and adding the HDF5 root dir, include dir and lib to the cmake with -D, but this did not help either...

�[0m[ 95%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Parallel/Applicator.cc.o
[ 95%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Parallel/MPIError.cc.o
[ 96%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Parallel/MPITransport.cc.o
[ 96%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Parallel/PabloIO.cc.o
[ 96%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Parallel/SerialTransport.cc.o
[ 97%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Utilities/FixVis.cc.o
[ 97%] Building CXX object synthesis/CMakeFiles/casa_synthesis.dir/Utilities/ThreadCoordinator.cc.o
[ 97%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/fgridft.f.o
[ 98%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/fmosaic.f.o
[ 98%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/fpbmos.f.o
[ 98%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/fpbwproj.f.o
[ 99%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/leakantso.f.o
[ 99%] Building Fortran object synthesis/CMakeFiles/casa_synthesis.dir/fortran/wprojgrid.f.o
Linking CXX shared library libcasa_synthesis.so
[ 99%] Built target casa_synthesis
Scanning dependencies of target lwimager
[100%] Building CXX object synthesis/CMakeFiles/lwimager.dir/apps/lwimager.cc.o
Linking CXX executable lwimager
�[91mCMakeFiles/lwimager.dir/apps/lwimager.cc.o: In function casa::HDF5Group::HDF5Group(casa::HDF5Object const&, casa::String const&, bool, bool)': lwimager.cc:(.text._ZN4casa9HDF5GroupC2ERKNS_10HDF5ObjectERKNS_6StringEbb[_ZN4casa9HDF5GroupC5ERKNS_10HDF5ObjectERKNS_6StringEbb]+0x7a): undefined reference tocasa::HDF5Group::init(int, casa::String const&, casa::String const&, bool, bool)'
�[0m�[91mcollect2: error: ld returned 1 exit status
�[0m�[91mmake[2]: *** [synthesis/lwimager] Error 1
�[0m�[91mmake[1]: *** [synthesis/CMakeFiles/lwimager.dir/all] Error 2
�[0m�[91mmake: *** [all] Error 2
�[0mThe command '/bin/sh -c cd ${INSTALLDIR}/casarest/build && make -j ${J}' returned a non-zero code: 2

**Comment by oonkjbr** [comment]
A possible solution for centos(7) is to remove the HDF5 install where hdf5 is obtained via wget from support.hdfgroup.org and instead install hdf5-devel via: (epel is needed for centos as the ubuntu libhdf5-dev does not have a a libhdf5-devel counterpart in centos)

sudo yum install –y epel-release
sudo yum-config-manager --enable epel
sudo yum install hdf5-devel

-> see also issue 6 #18

**Comment by tammojan** [comment]
Should be solved by f1d73d044676bb2a0c41bc28e3b07611748d8abc.
